src/Components/SearchInput.js

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import { withRouter } from "react-router";
```

Comment:
There's a hook from react-router-dom that is called `useHistory`, which is a better use case than this HOC (high order component). Check: https://reactrouter.com/web/api/Hooks